iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/RemoveHvdcLine.java

```diff
@@ -0,0 +1,116 @@
+
+    public RemoveHvdcLine(String hvdcLineId, List<String> shuntCompensatorsIds) {
+        this.hvdcLineId = Objects.requireNonNull(hvdcLineId);
+        this.shuntCompensatorsIds = Objects.requireNonNull(shuntCompensatorsIds);
+    }
+
+    @Override
+    public void apply(Network network, boolean throwException, ComputationManager computationManager, Reporter reporter) {
+        HvdcLine hvdcLine = network.getHvdcLine(hvdcLineId);
+        if (hvdcLine != null) {
+            HvdcConverterStation<?> hvdcConverterStation1 = hvdcLine.getConverterStation1();
+            HvdcConverterStation<?> hvdcConverterStation2 = hvdcLine.getConverterStation2();
+            hvdcLine.remove();
+            removedHvdcLineReport(reporter, hvdcLineId);
+            // Remove the Shunt compensators that represent the filters of the LCC
+            removeShuntCompensators(network, hvdcConverterStation1, hvdcConverterStation2, reporter);
+            removeConverterStations(hvdcConverterStation1, hvdcConverterStation2, reporter);
+        } else {
+            LOGGER.error("HVDC Line {} not found", hvdcLineId);
+            notFoundHvdcLineReport(reporter, hvdcLineId);
+            if (throwException) {
+                throw new PowsyblException("HVDC Line not found: " + hvdcLineId);
+            }
+        }
+    }
+
+    private void removeShuntCompensators(Network network,
+                                         HvdcConverterStation<?> hvdcConverterStation1,
+                                         HvdcConverterStation<?> hvdcConverterStation2,
+                                         Reporter reporter) {
+        if (!shuntCompensatorsIds.isEmpty()
+                && (isLccConverterStation(hvdcConverterStation1)
+                || isLccConverterStation(hvdcConverterStation2))) {
+
+            // Get Lcc voltage levels
+            List<VoltageLevel> lccVoltageLevel = network.getVoltageLevelStream()
+                    .filter(voltageLevel -> voltageLevel.getLccConverterStationStream()
+                            .anyMatch(lccConverterStation -> lccConverterStation.getId().equals(hvdcConverterStation1.getId())
+                                    || lccConverterStation.getId().equals(hvdcConverterStation2.getId())))
+                    .collect(Collectors.toList());
+
+            // removing shunt compensators
+            shuntCompensatorsIds.forEach(shuntCompensatorId -> {
+                // check whether the shunt compensator is connected to the same voltage level as the LCC
+                boolean isSameVoltageLevel = network.getVoltageLevelStream()
+                        .filter(voltageLevel -> lccVoltageLevel.contains(voltageLevel))
+                        .flatMap(voltageLevel -> voltageLevel.getConnectableStream(ShuntCompensator.class))
+                        .anyMatch(shuntCompensator -> shuntCompensator.getId().equals(shuntCompensatorId));
+
+                if (isSameVoltageLevel) {
```

Comment:
So the test can be simple here. Something like : 
VoltageLevel shuntCompensatorVL = network.getShuntCompensator(shuntCompensatorId).getTerminal().getVoltageLevel();
                if(shuntCompensatorVL == vl1 || shuntCompensatorVL == vl2)